### DISCUSSION TITLE
Using Playground v2.5

### DISCUSSION THREAD

**Discussion by tbuyle** [lllyasviel/Fooocus#2395] [Q&A] [answered]
I am trying to use the news [Playground v2.5](https://huggingface.co/playgroundai/playground-v2.5-1024px-aesthetic) but can't get it to work.

![image](https://github.com/lllyasviel/Fooocus/assets/467350/5bdf1f36-cf2d-47ec-8a66-8049e8ee181c)

I guess there are some important configuration I am missing. could this be about these EDM samplers ? Ot is it something else ?

**Comment by mashb1t** [comment]
> about these EDM samplers

yup, exactly. They're not using the standard way of generating images. I have the exact same issues and was wondering the same thing ^^

"We will also provide extensions for using our model with A1111 and ComfyUI in the near future"
~ https://playground.com/blog/playground-v2-5

plus see https://www.reddit.com/r/StableDiffusion/comments/1b1ht9i/comment/ksfqky3/

**Comment by tbuyle** [comment]
Here is hiw they did it in Comfy apparently...

https://github.com/comfyanonymous/ComfyUI/commit/d46583ecece5014f23f9f47f7952c8aecd8cc491

**Reply by mashb1t** [reply]
Great to know, thank you for the reference. Feel free to create a PR, my current prio is v2.2.0 as v2.2.0-rc1 has been in review for a few days now and i've left the ldm_patched core to lllyasviel, who might already work on implementing SD3 support.

**Reply by tbuyle** [reply]
I'd love to create the PR but I am not sure how to used that code... I could copy the SDXL_Playground_2_5 from Comfy in latent_formats.py and probably copy other snippets in other file that are similar to Comfy's. But I ahve no idea how these would need to be used / called in the rest of the code...

**Comment by PierrunoYT** [comment]
I don't like Comfy because it's not easy to use.

**Reply by newxhy** [reply]
me too，It cannot yet become a productivity tool

**Comment by mashb1t** [comment, marked as answer]
=> Implemented in https://github.com/lllyasviel/Fooocus/pull/3073, please test CFG so we can find the best one for the preset (tested with 2 and 3, both with pro/contra).

**Reply by ErmMonzon** [reply]
Playground v2.5 works best with CFG2, it's a decent middle ground. CFG 3 can look too saturated and over contrasty. Of course users can adjust, personally sometimes I'll even use CFG1.5 but 2 is a good default.
![PGv2 5 CFG](https://github.com/lllyasviel/Fooocus/assets/129614926/b12332c9-1375-40da-be04-a4186cf63134)

**Reply by graz68a** [reply]
Do you mean we shold set "CFG Mimicking from TSNR" to 2.5 ?

**Comment by wisdombox44** [comment]
Thank you, mashb1t, GREAT JOB! Playground 2.5 is amazing. My experience with Playground 2.5 has shown that it works best in speed performance. Once I got good results, I upscaled with the SDXL model, which gave me more detailed results. Playground 2.5 has a unique artistic style and cinematic view, while SDXL models are still better in face details. Combining the two gives me great results. Here are my first experiments.
![2024-06-06_21-32-33_1075-2b](https://github.com/lllyasviel/Fooocus/assets/123548315/980ce061-5a24-4480-a2a5-0f0e8a2ee171)
![2024-06-06_21-32-33_1075-3](https://github.com/lllyasviel/Fooocus/assets/123548315/35a11326-80df-4a9b-ae93-df6037fb5474)

**Reply by Alexdnk** [reply]
Try samplers **dpmpp_2s_ancestral** and **dpmpp_sde**, they give amazing results. But they require x2 generation time, unlike sampler **dpmpp_2m**. Prompt 'a man and a woman in front of an artistic background'+Fooocus Photograph+Artstyle Hyperrealism.
Thank you, @mashb1t!
dpmpp_2m:
![1](https://github.com/lllyasviel/Fooocus/assets/83111151/f0085f6c-e02f-4b5b-a8c9-e8607b5f9795)
dpmpp_2s_ancestral:
![2](https://github.com/lllyasviel/Fooocus/assets/83111151/e99b0653-959c-4ccd-aab6-94a285217b5c)
dpmpp_sde:
![3](https://github.com/lllyasviel/Fooocus/assets/83111151/98128b50-e2fc-46dd-b3de-22559f88793c)

**Reply by wisdombox44** [reply]
Looks beautiful, definetley playground 2.5 has artistic style.

**Reply by PeterJared** [reply]
I just downloaded again Fooocus to be sure no errors have been added from my side and  used the current playground 2.5 preset (it downloaded the needed files). But I get renders with tons of artifacts:
![image](https://github.com/lllyasviel/Fooocus/assets/49351356/8b849902-2f95-45a6-9f7e-699128798913)
![image](https://github.com/lllyasviel/Fooocus/assets/49351356/3b970a28-0939-451d-ac6a-7ac5ca792c14)

**Reply by Rogala** [reply]
I can't understand what's wrong, I try playground, but I get strange results.
All settings are standard I change only CFG from 1-4, nothing good comes out.
I try other modes all seem to be more or less normal, what is the problem why such poor quality?
![2024-06-10_143917](https://github.com/lllyasviel/Fooocus/assets/7437582/31492877-8abf-4f1d-a9d1-1ef36d4545f6)

**Reply by PeterJared** [reply]
It seems to me we are missing some of the parameters used in the examples included in this page. The current  defaults in Fooocus for Playground give those wrong results and playing around with them don't improve anything.

**Reply by Alexdnk** [reply]
@Rogala @PeterJared try set this sampler. I think this is best for pg. Well it's pg, it is what it is.
![1](https://github.com/lllyasviel/Fooocus/assets/83111151/4bab11c2-67ad-41e3-ac42-6257365b2f77)

**Reply by PeterJared** [reply]
can you send a image that stores the metadata? It will be the easier way to get all parameters. Those settings above are not yet enough.

**Reply by Rogala** [reply]
There is no change, the picture is very bad
```
{
"prompt": "Earth",
"negative_prompt": "",
"prompt_expansion": "Earth, modern, intricate, elegant, highly detailed, extremely chic background, professional fine detail, bright colors, cool dynamic dramatic color, cute, beautiful, delicate, advanced, full cinematic, symmetry, great composition, passionate, rich deep vivid, ambient, sharp focus, artistic, expressive, attractive, thoughtful, thought, iconic, best, perfect, stunning, creative, positive",
"styles": "['Fooocus V2', 'Fooocus Enhance', 'Fooocus Sharp']",
"performance": "Speed",
"resolution": "(1024, 1024)",
"guidance_scale": 3,
"sharpness": 2,
"adm_guidance": "(1.5, 0.8, 0.3)",
"base_model": "playground-v2.5-1024px-aesthetic.fp16.safetensors",
"refiner_model": "None",
"refiner_switch": 0.5,
"clip_skip": 2,
"sampler": "dpmpp_2s_ancestral",
"scheduler": "edm_playground_v2.5",
"vae": "Default (model)",
"seed": "2394789218168369462",
"metadata_scheme": false,
"version": "Fooocus v2.4.3 (mashb1t)"
}
```
![2024-06-10_18-36-30_1958](https://github.com/lllyasviel/Fooocus/assets/7437582/008f4637-570f-4880-b508-54fff7f70a38)

**Comment by tbuyle** [comment]
Hadn't the opportunity to test it yet. But will try to find some time this week-end.

👍 Thanks @mashb1t mashb1t for the amazing work (on this and on the many others issues and requests)

**Comment by PeterJared** [comment]
Is there a guide to test this? I have no idea how to start.

**Reply by mashb1t** [reply]
You can find additional resources here:
- https://playground.com/blog/playground-v2-5
- https://huggingface.co/playgroundai/playground-v2.5-1024px-aesthetic
- https://arxiv.org/abs/2402.17245
- https://civitai.com/models/325263/playground-ais-playground-v25-1024px

**Comment by ErmMonzon** [comment]
Great to see PGv2.5 implemented in Fooocus! Some things to consider to get you started. 

**Guidance Scale** This is personal preference but typically 1-3 is ideal, personally I use 2 mostly. It's very sensitive so only increase by 0.5 increments.
In the advanced tab select "developer debug mode"
![image](https://github.com/lllyasviel/Fooocus/assets/129614926/2bc3dfff-d57b-4551-b52a-6d4cfb80cb0a)

Under sampler you have to change the scheduler to edm_playground_v2.5
Sampler, dpmpp_2m is recommended but you can always experiment. dpmpp_2m_sde_gpu seems to work well, euler and euler_ancestral will give you slightly softer results
![image](https://github.com/lllyasviel/Fooocus/assets/129614926/3de7771f-e615-4eb2-937c-b694dceecf68)

![PGv2 5 CFG](https://github.com/lllyasviel/Fooocus/assets/129614926/6a6f9dfa-b1bc-4bd2-b90b-f4ac0491b4f2)

Close up portraits will give great details but full to half body will likely need some inpainting for faces. Fooocus inpainting model works very well to correct these limitations.

This model has a very different aesthetic, it does have photorealistic capabilities but I find its a very creative model with it's own style and look. 

Hope this helps!

**Reply by mashb1t** [reply]
https://github.com/lllyasviel/Fooocus/pull/3136 reduces the CFG from 3 to 2 and removes some styles from the preset with the goal to optimize the generated image output. Please check out the latest main branch / restart Fooocus.

**Reply by graz68a** [reply]
sorry where should I reduce CFG  in Fooocus exactly ?

**Comment by ILikeToasters** [comment]
After a lot of experimenting with Playground I decided to try some image prompts and inpainting.... It seems those do not work with Playground. Am I getting expected results when it comes to this or am I missing something? As for the CFG I do agree that 2 is probably the best default.

**Reply by CantainPike** [reply]
Upscale 2X with playground also does not give you good results. But if you switch the model to any photo-realistic SDXL, when you upscale 2X, you will get great results. In my personal experience, upscale worked with Leosams Hello World. I liked Playground2.5's color-rich cinematic aesthetics.

**Comment by realies** [comment]
why is this model not capable of variation/upscaling and inpainting?

**Reply by mashb1t** [reply]
@realies
- variation: no issues, works
- upscaling: it has not been trained on high res inages, but should work to a certain degree. You may have to fiddle with the upscaling denoise overwrite setting for individual preference.
- inpainting: Fooocus uses an so called inlainting engine, a model which has been trained for SDXL. It does produce very noisy images for Playground / Pony models as their base data is completely different. This is the reason why the inpaint engine is set to None by default when selecting their corresponding preset. (also see https://github.com/lllyasviel/Fooocus/issues/3364)
